obsplus/structures/grid.py

```diff
@@ -0,0 +1,1323 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Created on Sat Nov 18 10:13:03 2017
+
+@author: shawn
+"""
+
+from copy import copy
+from typing import Iterable, Callable
+import os.path
+import warnings
+from numbers import Integral
+
+import numpy as np
+import pandas as pd
+import matplotlib.pyplot as plt
+import matplotlib.gridspec as gridspec
+from scipy.interpolate import griddata
+
+from obsplus.utils import read_file
+
+try:
+    import pyproj
```

Comment:
Making this optional is good, pyproj is a heavy dependency to require. I wonder if it might be better to move this the start of the `conver_coords` function and raise an exception there because it is the only thing that uses it. We don't want to get this warning every time we just want to use a grid for example.